### DISCUSSION TITLE
Batch correction

### DISCUSSION THREAD

**Discussion by jblich870** [GreenleafLab/ArchR#1695] [Feature Requests]
Hi,

Might it be possible to implement a CCA-based batch correction method within ArchR?

I have 3 time points across 2 replicate batches (6 samples total). Harmony works well for integrating time points together, but fails to fully integrate the replicate batches, despite correcting for multiple covariates and playing with lambda and theta values etc. Since I am not interested in the batch signature, it is problematic for downstream analyses that the replicate datasets can't be properly integrated (particularly pseudotime trajectory-based analyses).

The CCA-based batch correction method implemented in Seurat/Signac (FindIntegrationAnchors followed by IntegrateData) works well for correcting batch effects between technical replicates as it assumes a high degree of shared cell types between datasets. Might it be possible to incorporate this, or similar, into ArchR? E.g. one could use CCA to batch correct replicates and then Harmony to merge time points together, or alternatively, first use Harmony to merge time points, then CCA to batch correct the 2 sets of merged time points.

I think my experimental setup is fairly common and I suspect there would be appetite for this (this has been raised before, for example #153, #179, #786, #768, #582).

Would be very grateful to hear your thoughts.
Many thanks,
J

**Comment by rcorces** [comment]
This is a good suggestion. I would be happy to review a pull request if anyone wants to attempt to implement this. We're currently bogged down with other updates but I will otherwise add this to our list.

**Reply by jblich870** [reply]
Many thanks for your reply. I'd be happy to create a pull request but seem unable to create a new branch? Would be grateful if you could point me in the right direction (never done this before).
Thanks,
J

**Reply by rcorces** [reply]
more or less - https://opensource.com/article/19/7/create-pull-request-github
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

**Reply by jblich870** [reply]
Thanks: https://github.com/GreenleafLab/ArchR/pull/1737
